# HG changeset patch
# User Rik <rik@octave.org>
# Date 1522709701 25200
#      Mon Apr 02 15:55:01 2018 -0700
# Branch stable
# Node ID 69cbaa17978014740e8f65e6cc349175fb660411
# Parent  2b8442c890d8df75ea736db8b1b3112c6b782273
test: Temporarily workaround delete() failure in BIST tests (bug #53513).

* graphics.cc: Add a 5ms pause after deleting uicontextmenu so that changes can
propagate to parent figure.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -9427,16 +9427,17 @@ uicontextmenu::~uicontextmenu (void)
 %!   hctx1 = uicontextmenu ("parent", hf);
 %!   hctx2 = uicontextmenu ("parent", hf);
 %!   set (hf, "uicontextmenu", hctx2);
 %!   set (hax, "uicontextmenu", hctx2);
 %!   assert (get (hf, "uicontextmenu"), hctx2);
 %!   assert (get (hax, "uicontextmenu"), hctx2);
 %!   assert (get (hf, "children"), [hctx2; hctx1; hax]);
 %!   delete (hctx2);
+%!   pause (.005);
 %!   assert (get (hf, "uicontextmenu"), []);
 %!   assert (get (hax, "uicontextmenu"), []);
 %!   assert (get (hf, "children"), [hctx1; hax]);
 %!   set (hf, "uicontextmenu", hctx1);
 %!   assert (get (hf, "uicontextmenu"), hctx1);
 %!   set (hf, "uicontextmenu", []);
 %!   assert (get (hf, "uicontextmenu"), []);
 %!   assert (get (hf, "children"), [hctx1; hax]);
